### PR TITLE
SnabbWall introduction article

### PR DESCRIPTION
**Pull request by aperezdc**

### CODE REVIEW

File: 2015/introducing-snabbwall.markdown [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+<figure style="text-align:center">
+  ![](http://snabbwall.org/images/igalia-logo.png)
+  <span style="margin-left:20px">&nbsp;</span>
+  ![](http://snabbwall.org/images/nlnet-logo.gif)
+</figure>
+
+The one-minute introduction to Snabb
+------------------------------------
+
+Snabb works by taking over network interfaces, bypassing the kernel drivers, and using its own. When used to implement a certain network functinoality, what your (Snabb) program “sees” is no more than streams of raw packets. Gone are the TCP/IP stack, BSD sockets, and all the other functionality provided by the operating system kernel to user space programs. Heck, you even need to roll your [own](https://github.com/SnabbCo/snabbswitch/tree/master/src/apps/intel) [drivers](https://github.com/SnabbCo/snabbswitch/tree/master/src/apps/solarflare)! There is nothing in there but the hardware waiting to have streams of electrons blasted through it. But things go *real* fast in there without the overhead of a traditional network stack, so despite the void it is a good place to massage packets and pass them along.
```
Comment (dpino) [inline 2015/introducing-snabbwall.markdown]: Snabb is the name of Luke's company [1], the name of the software is Snabb Switch.
Typo: functinoality.
Doubt: Did you mean "message packets"?

[1] http://snabb.co/

File: 2015/introducing-snabbwall.markdown [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+It is fortunate that empty spaces do not last long as such: Snabb comes with a set of [built-in applications](https://github.com/SnabbCo/snabbswitch/tree/master/src/apps), which can be combined in different ways to achieve our (networking-related) goals. For example, we could pick the [bridge application](https://github.com/SnabbCo/snabbswitch/tree/master/src/apps/bridge), combine it with the driver for our network interfaces, and *(bam!)* we have a super fast Ethernet bridge. And even more.
```
Comment (dpino) [inline 2015/introducing-snabbwall.markdown]: If you don't want to type Snabb Switch all the time, you can introduce its short form, "SnS", the first time you mention it.

File: 2015/introducing-snabbwall.markdown [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+```
+
+Well, that was not long. But you can see where this is going: if we are serious about making a full–fledged firewall, our cute little program is going to mutate into something hairy. We would like to have something like [netfilter](http://netfilter.org/) or, even better, something neater which allows to specify rules in a higher level syntax like [OpenBSD's Packet Filter](http://www.openbsd.org/faq/pf/filter.html), package it as a Snabb Switch application, and have it ready for reusing.
+
+
+Dear readers: meet SnabbWall
+----------------------------
+
+SnabbWall is —will be, when completed— a set of Snabb Switch *applications* and *programs*. Applications are reusable in your own designes, like every other Snabb Switch application.
```
Comment (dpino) [inline 2015/introducing-snabbwall.markdown]: Typo: s/designes/designs/

File: 2015/introducing-snabbwall.markdown [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+There are Free Software solutions which already provide DPI functionality. After evaluating the candidates, it was decided that SnabbWall (in particular, the L7 Spy application) would use [nDPI](http://www.ntop.org/products/ndpi/). It is a well-maintained, reliable piece of software, with a C API which lends itself to be wrapped using [LuaJIT's FFI](http://luajit.org/ext_ffi_api.html). It has a long story —for a piece of software— as it started life as OpenDPI, which itself was a trimmed down version of a commercial product, and along its multiple lifes it has been tweaked to the point that nDPI provides the best protocol/application detection of the pack.
```
Comment (dpino) [inline 2015/introducing-snabbwall.markdown]: Typo: s/lifes/lives/

**Comment by dpino** [comment]
Reviewed. LGTM. Please, fix typos (or better to pass a spellchecker).

File: 2015/introducing-snabbwall.markdown (hunk 1 above)
Comment (aperezdc) [inline 2015/introducing-snabbwall.markdown]: Typo fixed, and also fixed “suprise” in the first paragraph.

I mean “massage packets”, as written. As in “a packet comes, your code gives it a massage _(meaning: ‘manipulates it’)_, and then it is sent somewhere else”.

File: 2015/introducing-snabbwall.markdown (hunk 2 above)
Comment (aperezdc) [inline 2015/introducing-snabbwall.markdown]: Yeah, I think that's a good idea. Thanks for mentioning it.

File: 2015/introducing-snabbwall.markdown (hunk 3 above)
Comment (aperezdc) [inline 2015/introducing-snabbwall.markdown]: ACK.

File: 2015/introducing-snabbwall.markdown (hunk 4 above)
Comment (aperezdc) [inline 2015/introducing-snabbwall.markdown]: ACK.

**Comment by aperezdc** [comment]
I will fix the typos and make “Snabb” (the name of the company) is not used, and then merge this to `master` to get the article published.

Thanks a lot for the review, @dpino!

**Pull request merged**